### PR TITLE
[SMALLFIX] bugfix for ls when a user does not have permission to access a child dir 

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java [hunk 1]
```diff
@@ -975,7 +975,7 @@ private void listStatusInternal(LockedInodePath currInodePath,
                                   List<FileInfo> statusList)
       throws FileDoesNotExistException, UnavailableException, AccessControlException {
     Inode<?> inode = currInodePath.getInode();
-    if (inode.isDirectory()) {
+    if (inode.isDirectory() && descendantType != DescendantType.NONE) {
```
Comment (bf8086) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java]: You can remove the check from line 990.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by yuzhu** [comment]
@gpang before this change, if the user tries to `ls` on a directory that has a subdirectory it does not have execute permission on, it will fail.

**Pull request merged**